src/mappings/_utils.ts

```diff
@@ -127,6 +130,156 @@ export async function friendlyAmount(currency: Currency, amount: number): Promis
+    switch(currency.isTypeOf) {
+        case 'NativeToken':
+            switch (currency.token) {
+                case 'KINT':
+                case 'KSM':
+                    return 12
+                case 'INTR':
+                case 'DOT':
+                    return 10
+                case 'KBTC':
+                case 'IBTC':
+                    return 8
+                default:
+                    return 0
+            }
+        case 'ForeignAsset':
+            const details = await getForeignAsset(currency.asset)
+            return details.decimals;
+        default:
+            return 0;
+    }
+}
+
+type CurrencyType = Bitcoin | Kintsugi | Kusama | Interlay | Polkadot;
+
+function mapCurrencyType(currency: Currency): CurrencyType {
+    switch(currency.isTypeOf) {
+        case 'NativeToken':
+            switch(currency.token) {
+                case 'KINT':
+                    return Kintsugi;
+                case 'KSM':
+                    return Kusama;
+                case 'INTR':
+                    return Interlay;
+                case 'DOT':
+                    return Polkadot;
+                case 'KBTC':
+                    return KBtc;
+                case 'INTR':
+                    return InterBtc;
+            }
+        case 'ForeignAsset':
+            return Bitcoin;
+        default:
+            throw new Error(`Unsupported currency type: ${currency.isTypeOf}`);
+    }
+}
+
+type OracleRate = {
+    btc: Big;
+    usdt: Big;
+}
+
+/* This function is used to calculate the exchange rate for a given currency at
+a given time.
+*/
+export async function getExchangeRate(
+    ctx: Ctx,
+    timestamp: number,
+    currency: Currency,
+    amount: number
+): Promise<OracleRate>  {
+    const mappedCurrency = mapCurrencyType(currency);
+    let baseMonetaryAmount
+    let searchBlock = currency.toJSON();
+
+    if (mappedCurrency === KBtc || mappedCurrency === InterBtc) {
+        baseMonetaryAmount = newMonetaryAmount(Big(1e8), Bitcoin)
+    }
+    else {
+        let lastUpdate = await ctx.store.get(OracleUpdate, {
+            where: { 
+                id: Like(`%${JSON.stringify(searchBlock)}`),
+                timestamp: LessThanOrEqual(new Date(timestamp)),
+            },
+            order: { timestamp: "DESC" },
+        });
+        console.log(lastUpdate?.timestamp);
+        if (lastUpdate === undefined) {
+            ctx.log.warn(
+                `WARNING: no price registered by Oracle for ${JSON.stringify(searchBlock)} at timestamp ${new Date(timestamp)}. Fetching first value.`
+            );
+            lastUpdate = await ctx.store.get(OracleUpdate, {
+                where: { 
+                    id: Like(`%${JSON.stringify(searchBlock)}`),
+                },
+                order: { timestamp: "ASC" },
+            });
+        }
+        const lastPrice = new Big((Number(lastUpdate?.updateValue) || 0) / 1e10);
```

Comment:
The division by `1e10` seems arbitrary to an outsider. Can we please add a comment as to why we're dividing by that number?

Reply:
done